### PR TITLE
refactor: switch to ESM

### PR DESCRIPTION
**Pull request by king-11**
**What is the purpose of this pull request? (put an "X" next to item)**

- [x] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

**What changes did you make? (Give an overview)**

- chore: update imports and export index.js
- chore: update imports in tests
- chore: bump bitfield for esm
- chore: update package.json for esm

**Which issue (if any) does this pull request address?**

- https://github.com/webtorrent/webtorrent/issues/2177

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by king-11** [comment]
@DiegoRBaquero please take a look

File: index.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-const debug = require('debug')('bittorrent-protocol')
-const randombytes = require('randombytes')
-const sha1 = require('simple-sha1')
-const speedometer = require('speedometer')
-const stream = require('readable-stream')
-const RC4 = require('rc4')
+import bencode from 'bencode'
+import BitField from 'bitfield'
+import crypto from 'crypto'
+import debugFunction from 'debug'
```
Comment (DiegoRBaquero) [inline index.js:5]:
```suggestion
import Debug from 'debug'
```

File: index.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+import randombytes from 'randombytes'
+import RC4 from 'rc4'
+import stream from 'readable-stream'
+import sha1 from 'simple-sha1'
+import speedometer from 'speedometer'
+import arrayRemove from 'unordered-array-remove'
+
+const debug = debugFunction('bittorrent-protocol')
```
Comment (DiegoRBaquero) [inline index.js:13]:
```suggestion
const debug = Debug('bittorrent-protocol')
```

**Comment by king-11** [comment]
@DiegoRBaquero made the suggested changes

**Pull request merged**